docs/docs/01-ibc/06-channel-upgrades.md

```diff
@@ -11,6 +11,46 @@ slug: /ibc/channel-upgrades
 Learn how to upgrade existing IBC channels.
 :::
 
+Channel upgradeability is an IBC-level protocol that allows chains to leverage new channel features without having to create new channels or perform a network-wide upgrade. 
+
+Prior to this feature, developers who wanted to update an application module or add a middleware to their application flow would need to negotiate a new channel in order to use the updated application feature/middleware, resulting in a loss of the accumulated state/liquidity, token fungibility (as the channel would have been encoded in the IBC denom), and any other larger network effects of losing usage of the existing channel from relayers monitoring, etc.
+
+With channel upgradeability, applications will be able to implement features such as but not limited to: [including a memo field in the packet data for fungible tokens](https://github.com/cosmos/ibc/pull/842), adding [denom metadata to tokens](https://github.com/cosmos/ibc/discussions/719), or utilizing the [fee middleware](https://github.com/cosmos/ibc/tree/main/spec/app/ics-029-fee-payment), all while maintaining the channels on which they currently operate.
+
+This document outlines the channel upgrade feature, and the multiple steps used in the upgrade process.
+
+## Channel Upgrade Handshake
+
+Channel upgrades will be initialized using a handshake process that is designed to be similar to the standard connection/channel opening handshake.
+
+```go
+interface ChannelEnd {
+  state: ChannelState
+  ordering: ChannelOrder
+  counterpartyPortIdentifier: Identifier
+  counterpartyChannelIdentifier: Identifier
+  connectionHops: [Identifier]
+  version: string
+  upgradeSequence: uint64
+}
```

Comment:
Instead of the interface from the spec, should we use here the proto or go definition?